src/stirling/source_connectors/socket_tracer/conn_tracker.h

```diff
@@ -271,9 +271,33 @@ class ConnTracker : NotCopyMoveable {
     CONN_TRACE(2) << absl::Substitute("req_frames=$0 resp_frames=$1", req_frames.size(),
                                       resp_frames.size());
 
-    protocols::RecordsWithErrorCount<TRecordType> result =
-        protocols::StitchFrames<TRecordType, TFrameType, TStateType>(&req_frames, &resp_frames,
-                                                                     state_ptr);
+    protocols::RecordsWithErrorCount<TRecordType> result;
+
+    // If this protocol doesn't support streams, we call StitchFrames with just the deque.
+    // If it does, we use a map of stream ID to deque.
+    if constexpr (TProtocolTraits::stream_support ==
+                  protocols::BaseProtocolTraits<TRecordType>::UseStream) {
+      using TKey = typename TProtocolTraits::key_type;
+      std::map<TKey, std::deque<TFrameType>> requests;
+      std::map<TKey, std::deque<TFrameType>> responses;
+      // TODO(@benkilimnik): Hard code the stream for now. Populate the map in a future PR.
+      requests[0] = std::move(req_frames);
+      responses[0] = std::move(resp_frames);
+      result = protocols::StitchFrames<TRecordType, TKey, TFrameType, TStateType>(
+          &requests, &responses, state_ptr);
+      // TODO(@benkilimnik): Update req and resp frame deques to match maps for now. Populate maps
+      // during parsing in a future PR.
+      req_frames.clear();
+      resp_frames.clear();
+      for (auto& [_, frames] : requests) {
+        for (auto& frame : frames) {
+          req_frames.push_back(frame);
+        }
+      }
```

Comment:
Was it intentional to only copy the frames from the request map?

Reply:
For the CQL stitcher and a few other protocols, the response deque is cleared at the end of `StitchFrames`, so I don't think we need to copy the frames from the responses map. In this PR, only CQL has been set to use the map interface.

You're right that this might be fail if other protocols used the map, though this should be resolved once 3/3 gets in and the maps are populated earlier during parsing.

Reply:
While this code should be short lived, I think it would be best to avoid relying on any implicit behavior within the stitcher. So my preference would be to iterate over both of them.

Reply:
Added in most recent commit